### PR TITLE
Feature/cronjob

### PR DESCRIPTION
**Pull request by xtuchyna**
## This introduces a breaking change

- [ ] Yes
- [ X ] No

## This Pull Request implements

Templates for openshift's imagestream, build and cronjob
This PR also adds environment variable option to set cli in a specific way (create knowledge and which specific projects to inspect)

### CODE REVIEW

File: openshift/cronJob-template.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              - name: THOTH_CEPH_KEY_ID
+                valueFrom:
+                  secretKeyRef:
+                    name: srcopsmetrics-secrets
+                    key: thothCephKeyID
+              - name: PYTHONPATH
+                value: "."
+              - name: PREFIX
+                value: data/thoth/dtuchyna
+              - name: HOST
```
Comment (pacospace) [inline openshift/cronJob-template.yaml]: These values can be taken from thoth ConfigMap :)

Comment (xtuchyna) [reply openshift/cronJob-template.yaml]: I created my own srcopsmetrics ConfigMap, is that right?

File: openshift/cronJob-template.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  schedule: "0 0 * * 0"
+  jobTemplate:
+    metadata:
+      name: srcopsmetrics
+    spec:
+      template:
+        spec:
+          containers:
+            - name: srcopsmetrics
+              image: docker-registry.default.svc:5000/dtuchyna-thoth-dev/srcopsmetrics
```
Comment (pacospace) [inline openshift/cronJob-template.yaml]: I don't think you have to provide all this, `srcopsmetrics` image will be pulled by default by the `docker-registry` in `dtuchyna-thoth-dev` if you deployed in that namespace.

File: openshift/cronJob-template.yaml (hunk 1 above)
Comment (pacospace) [reply openshift/cronJob-template.yaml]: I would reuse as much as possible what we have already if possible :)

File: openshift/cronJob-template.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  annotations:
+    thoth-station.ninja/template-version: 0.7.0
```
Comment (pacospace) [inline openshift/cronJob-template.yaml]: 0.1.0?

File: openshift/cronJob-template.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  annotations:
```
Comment (pacospace) [inline openshift/cronJob-template.yaml]: I would add the usual annotations we have in all templates. Have a look at: https://github.com/AICoE/sesheta/blob/ede0ecaaf57b25ee14018db9d0a53defa6a53782/openshift/cronJob-template.yaml#L10

File: openshift/cronJob-template.yaml (hunk 2 above)
Comment (xtuchyna) [reply openshift/cronJob-template.yaml]: when line 19 commented out, openshift console has problem
```The CronJob "srcopsmetrics" is invalid: spec.jobTemplate.spec.template.spec.containers[0].image: Required value```

Comment (pacospace) [reply openshift/cronJob-template.yaml]: Right, let's add a parameter in the template:

**Comment by pacospace** [comment]
Where are configmap and secret template?

File: openshift/cronJob-template.yaml (hunk 2 above)
Comment (xtuchyna) [reply openshift/cronJob-template.yaml]: added to all templates

**Comment by xtuchyna** [comment]
> Where is secret template?

I've added it just now, is it correct?

**Pull request merged**